### PR TITLE
deploy_helper: fix a bug when not defining release on state=clean

### LINKED ISSUE #1852: deploy_helper: missing release parameter for state=clean causes an error
**Summary**

The description of `release` parameter says that it is optional for `state=present` and required for `state=finalize`, but says nothing about `state=clean`. Executing a task with `state=clean` but without the `release` parameter causes an error

**Issue Type**

Bug Report

**Component Name**

deploy_helper

**Ansible Version**

```
ansible 2.10.5
  config file = /Users/maxim/Projects/XXX/ansible.cfg
  configured module search path = ['/Users/maxim/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/maxim/Library/Python/3.7/lib/python/site-packages/ansible
  executable location = /Users/maxim/Library/Python/3.7/bin/ansible
  python version = 3.7.3 (default, Apr 24 2020, 18:51:23) [Clang 11.0.3 (clang-1103.0.32.62)]
```

**Configuration**

_No response_

**OS / Environment**

MacOS Catalina (10.15) on control node, and Ubuntu 20.04 on managed node

**Steps To Reproduce**

```yaml
- community.general.deploy_helper:
    path: '{{ deploy_helper.project_path }}'
    state: clean
```

**Expected Results**

I don't know exactly how this should work, but I think we should add that the `release` parameter is required for `state=clean`, and handle this in code. Also the examples of use look wrong

**Actual Results**

```
The full traceback is:
Traceback (most recent call last):
  File "/home/XXX/.ansible/tmp/ansible-tmp-1613593157.5376189-33857-233175515274164/AnsiballZ_deploy_helper.py", line 102, in <module>
    _ansiballz_main()
  File "/home/XXX/.ansible/tmp/ansible-tmp-1613593157.5376189-33857-233175515274164/AnsiballZ_deploy_helper.py", line 94, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/home/XXX/.ansible/tmp/ansible-tmp-1613593157.5376189-33857-233175515274164/AnsiballZ_deploy_helper.py", line 40, in invoke_module
    runpy.run_module(mod_name='ansible_collections.community.general.plugins.modules.deploy_helper', init_globals=None, run_name='__main__', alter_sys=True)
  File "/usr/lib/python3.8/runpy.py", line 207, in run_module
    return _run_module_code(code, init_globals, run_name, mod_spec)
  File "/usr/lib/python3.8/runpy.py", line 97, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/ansible_community.general.deploy_helper_payload_zq3sjtgk/ansible_community.general.deploy_helper_payload.zip/ansible_collections/community/general/plugins/modules/deploy_helper.py", line 524, in <module>
  File "/tmp/ansible_community.general.deploy_helper_payload_zq3sjtgk/ansible_community.general.deploy_helper_payload.zip/ansible_collections/community/general/plugins/modules/deploy_helper.py", line 506, in main
  File "/tmp/ansible_community.general.deploy_helper_payload_zq3sjtgk/ansible_community.general.deploy_helper_payload.zip/ansible_collections/community/general/plugins/modules/deploy_helper.py", line 411, in remove_unfinished_link
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
fatal: [XXX]: FAILED! => {
    "changed": false,
    "module_stderr": "Shared connection to XXX closed.\r\n",
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/XXX/.ansible/tmp/ansible-tmp-1613593157.5376189-33857-233175515274164/AnsiballZ_deploy_helper.py\", line 102, in <module>\r\n    _ansiballz_main()\r\n  File \"/home/XXX/.ansible/tmp/ansible-tmp-1613593157.5376189-33857-233175515274164/AnsiballZ_deploy_helper.py\", line 94, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/home/XXX/.ansible/tmp/ansible-tmp-1613593157.5376189-33857-233175515274164/AnsiballZ_deploy_helper.py\", line 40, in invoke_module\r\n    runpy.run_module(mod_name='ansible_collections.community.general.plugins.modules.deploy_helper', init_globals=None, run_name='__main__', alter_sys=True)\r\n  File \"/usr/lib/python3.8/runpy.py\", line 207, in run_module\r\n    return _run_module_code(code, init_globals, run_name, mod_spec)\r\n  File \"/usr/lib/python3.8/runpy.py\", line 97, in _run_module_code\r\n    _run_code(code, mod_globals, init_globals,\r\n  File \"/usr/lib/python3.8/runpy.py\", line 87, in _run_code\r\n    exec(code, run_globals)\r\n  File \"/tmp/ansible_community.general.deploy_helper_payload_zq3sjtgk/ansible_community.general.deploy_helper_payload.zip/ansible_collections/community/general/plugins/modules/deploy_helper.py\", line 524, in <module>\r\n  File \"/tmp/ansible_community.general.deploy_helper_payload_zq3sjtgk/ansible_community.general.deploy_helper_payload.zip/ansible_collections/community/general/plugins/modules/deploy_helper.py\", line 506, in main\r\n  File \"/tmp/ansible_community.general.deploy_helper_payload_zq3sjtgk/ansible_community.general.deploy_helper_payload.zip/ansible_collections/community/general/plugins/modules/deploy_helper.py\", line 411, in remove_unfinished_link\r\nTypeError: unsupported operand type(s) for +: 'NoneType' and 'str'\r\n",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

### PR DESCRIPTION
**Pull request by ramondelafuente**
##### SUMMARY

Fixes #1852 

Do not try to `remove_unfinished_link` if no release was specifed.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/web_infrastructure/deploy_helper.py

##### ADDITIONAL INFORMATION

The deploy_helper module is useful when doing software deploys in versioned releases. It prepares the folder structure, and also cleans up failed deploys and/or older releases. 

In the case of calling `state=clean`, only the cleanup procedure is called. Specifying a release is optional in that case.

In my opinion, this bug stems from the fact that we are accessing `self.release` in the function `remove_unfinished_link` without passing that value as a parameter. So `self.release` is a hidden dependency of this method.

The current solution is to return early from within the `remove_unfinished_link` method - but a better solution would be to not depend on optional global state inside the module functions at all. `self.unfinished_filename` is also referenced a lot from within different methods (but since that has a default value in all cases, it does not cause problems currently).

I did not want to make this change larger than needed.

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Looks good! Can you add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)?

**Comment by ramondelafuente** [comment]
> Looks good! Can you add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)?

I think this should do it?

File: changelogs/fragments/1852-deploy-helper-fix-state-is-clean-without-release.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - deploy_helper - Allow ``state=clean`` to be used without defining a ``release`` (https://github.com/ansible-collections/community.general/issues/1852).
```
Comment (felixfontein) [inline changelogs/fragments/1852-deploy-helper-fix-state-is-clean-without-release.yaml:2]:
```suggestion
  - deploy_helper - allow ``state=clean`` to be used without defining a ``release`` (https://github.com/ansible-collections/community.general/issues/1852).
```

**Comment by felixfontein** [comment]
@ramondelafuente thanks! I've added a small suggestion (the main part of the fragment should always start lower-case). With that, this is good for merging!

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@ramondelafuente thanks for fixing this!

**Pull request merged**